File: lab3/Lab03_1/app/src/main/java/vn/edu/uit/lab03_1/StudentAdapter.java
package vn.edu.uit.lab03_1;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class StudentAdapter extends RecyclerView.Adapter<StudentAdapter.StudentViewHolder> {

    private List<Student> studentList;
    private final OnStudentClickListener clickListener;

    public interface OnStudentClickListener {
        void onItemClick(Student student);
        void onItemLongClick(Student student);
    }

    public StudentAdapter(List<Student> studentList, OnStudentClickListener clickListener) {
        this.studentList = studentList;
        this.clickListener = clickListener;
    }

    public void updateData(List<Student> newList) {
        this.studentList = newList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public StudentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_student, parent, false);
        return new StudentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull StudentViewHolder holder, int position) {
        Student student = studentList.get(position);
        holder.bind(student, clickListener);
    }

    @Override
    public int getItemCount() {
        return studentList.size();
    }

    static class StudentViewHolder extends RecyclerView.ViewHolder {
        private final TextView tvName;
        private final TextView tvInfo;

        public StudentViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tv_student_name);
            tvInfo = itemView.findViewById(R.id.tv_student_info);
        }

        public void bind(final Student student, final OnStudentClickListener listener) {
            tvName.setText(student.getName());
            String infoText = "MSSV: " + student.getStudentId() + " | Lớp: " + student.getClassName();
            tvInfo.setText(infoText);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onItemClick(student);
                    }
                }
            });

            itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View v) {
                    if (listener != null) {
                        listener.onItemLongClick(student);
                        return true;
                    }
                    return false;
                }
            });
        }
    }
}

File: lab3/lab3_2/app/src/main/java/com/example/lab3_2/StudentAdapter.kt
package com.example.lab3_2

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class StudentAdapter(private val students: List<Student>) : RecyclerView.Adapter<StudentAdapter.StudentViewHolder>() {

    private var onItemClickListener: ((Student) -> Unit)? = null

    fun setOnItemClickListener(listener: (Student) -> Unit) {
        onItemClickListener = listener
    }

    class StudentViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvName: TextView = view.findViewById(R.id.tvName)
        val tvMssv: TextView = view.findViewById(R.id.tvMssv)
        val tvLop: TextView = view.findViewById(R.id.tvLop)
        val tvDiem: TextView = view.findViewById(R.id.tvDiem)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StudentViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_student, parent, false)
        return StudentViewHolder(view)
    }

    override fun onBindViewHolder(holder: StudentViewHolder, position: Int) {
        val student = students[position]
        holder.tvName.text = student.name
        holder.tvMssv.text = "MSSV: ${student.mssv}"
        holder.tvLop.text = "Lớp: ${student.lop}"
        holder.tvDiem.text = String.format("%.1f", student.diem)

        holder.itemView.setOnClickListener {
            onItemClickListener?.invoke(student)
        }
    }

    override fun getItemCount(): Int = students.size
}
